templates/consul_catalog.tmpl

```diff
@@ -1,12 +1,33 @@
-[backends]{{range .Nodes}}
+[backends]
+{{range .Nodes}}
     [backends.backend-{{getBackend .}}.servers.server-{{.Node.Node | replace "." "-"}}-{{.Service.Port}}]
-    url = "http://{{.Node.Address}}:{{.Service.Port}}"
+    	url = "http://{{.Node.Address}}:{{.Service.Port}}"
```

Comment:
Maybe just this does the trick (waiting for build and setting up tests over here):

    diff --git a/templates/consul_catalog.tmpl b/templates/consul_catalog.tmpl
    index 4e86f32..1cdbeeb 100644
    --- a/templates/consul_catalog.tmpl
    +++ b/templates/consul_catalog.tmpl
    @@ -1,7 +1,12 @@
     [backends]
     {{range .Nodes}}
         [backends.backend-{{getBackend .}}.servers.server-{{.Node.Node | replace "." "-"}}-{{.Service.Port}}]
    +      {{$scheme := getAttribute "protocol" .Service.Tags}}
    +      {{if $scheme}}
    +      url = "{{$scheme}}//{{.Node.Address}}:{{.Service.Port}}"
    +      {{else}}
         	url = "http://{{.Node.Address}}:{{.Service.Port}}"
    +      {{end}}
             {{$weight := getAttribute "weight" .Service.Tags}}
             {{with $weight}}
                 weight = {{$weight}}
    @@ -30,4 +35,4 @@
       passHostHeader = false
         [frontends.frontend-{{.ServiceName}}.routes.route-host-{{.ServiceName}}]
         rule = "{{getFrontendValue .ServiceName}}"
    -{{end}}
    \ No newline at end of file
    +{{end}}